packages/@coorpacademy-components/src/molecule/dashboard/cards-list/index.js

```diff
@@ -175,77 +167,68 @@ class CardsList extends React.Component {
       const leftBound = this.leftBound();
       const rightBound = this.rightBound();
 
-      const skip = this.getPossiblePositions((el, index) => {
-        return el + getOr(0, [index, 'scrollWidth'], this.cards) <= leftBound;
+      const skip = this.getPossiblePositions().filter((el, index) => {
+        return el + this.getScrollWidth(this.cards[index]) <= leftBound;
       }).length;
-      const limit = this.getPossiblePositions((el, index) => {
-        return el + getOr(0, [index, 'scrollWidth'], this.cards) > leftBound && el < rightBound;
+      const limit = this.getPossiblePositions().filter((el, index) => {
+        return el + this.getScrollWidth(this.cards[index]) > leftBound && el < rightBound;
       }).length;
       onScroll(skip, limit);
     }
   }
 
   handleOnLeft() {
-    const currentScrollPos = this.leftBound();
-    if (currentScrollPos <= 0) {
+    const {actualPage} = this.state;
+    if (actualPage === 1) {
       return;
     }
-
-    this.scrollTo(currentScrollPos, true);
+    this.scrollTo(actualPage - 1);
   }
 
   handleOnRight() {
-    const currentScrollPos = this.leftBound();
-    if (currentScrollPos >= this.maxLeftBound()) {
+    const {actualPage} = this.state;
+    if (actualPage === this.maxPages()) {
       return;
     }
-
-    this.scrollTo(currentScrollPos, false);
+    this.scrollTo(actualPage + 1);
   }
 
-  scrollTo(currentScrollPos, toPrevious) {
-    const possiblePositions = this.getPossiblePositions(el => {
-      return toPrevious ? el < currentScrollPos : el > currentScrollPos + this.wrapperWidth();
-    });
-
-    if (!toPrevious) {
-      possiblePositions.push(this.cardsWidth());
-    }
+  scrollTo(page) {
+    const indexOfnextFirstCard = this.getPossiblePages().indexOf(page);
+    const nextPosition = sumBy('scrollWidth', this.cards.slice(0, indexOfnextFirstCard));
+    this.cardsWrapper.scrollLeft = nextPosition;
+    this.updatePages(page);
+  }
 
-    if (possiblePositions.length === 0) {
-      return;
+  updatePages(event_) {
+    const isResize = typeof event_ === 'object' && event_.type === 'resize';
+    const {actualPage} = this.state;
+    let nextState;
```

Comment:
plutot que nextState en let, je mettrais  `actualPage`, ce  qui  simplifirait l'affectation  dans les branche,  
et duu coup, ligne 220:

```js
if (!isEqual(this.state.actualPage, actualPage)) this.setState({actualPage});
```
(avec éventuelle protection `this.state` not an object)